control_plane/src/storage.rs

```diff
@@ -199,23 +199,42 @@ impl PageServerNode {
         bail!("pageserver failed to start in {} seconds", RETRIES);
     }
 
-    pub fn stop(&self) -> anyhow::Result<()> {
+    pub fn stop(&self, immediate: bool) -> anyhow::Result<()> {
         let pid = read_pidfile(&self.pid_file())?;
         let pid = Pid::from_raw(pid);
-        if kill(pid, Signal::SIGTERM).is_err() {
-            bail!("Failed to kill pageserver with pid {}", pid);
+        if immediate {
+            println!("Stop pageserver immediately");
+            if kill(pid, Signal::SIGQUIT).is_err() {
+                bail!("Failed to stop pageserver with pid {}", pid);
+            }
+        } else {
+            println!("Stop pageserver gracefully");
+            if kill(pid, Signal::SIGTERM).is_err() {
```

Comment:
IMO would be helpful to include the info about an error that had caused the issue.

I would write this `if` statement as `kill(pid, Signal::SIGTERM).with_context(|| "Failed to kill pageserver with pid {}", pid)?` and same for the rest here: this way, we not only return the `Failed to kill pageserver with pid ...` error, but its cause also.